### PR TITLE
Move testing to Github Actions

### PR DESCRIPTION
**Pull request by nicoleepp**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -42,7 +42,7 @@
         "License :: OSI Approved :: MIT License",
         "Operating System :: OS Independent",
         "Programming Language :: Python",
-        "Programming Language :: Python :: 3.7",
+        "Programming Language :: Python :: 3.8",
         "Programming Language :: Python :: 3.8",
```
Comment (fchorney) [inline setup.py:46]:
```suggestion
        "Programming Language :: Python :: 3.9",
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        run: |
+          python -m pip install --upgrade pip
+          pip install tox tox-gh-actions
+      # Run Tests
+      - name: Run Python Tests
+        run: tox
+
+      - name: Show items
+        run: ls -la
+
```
Comment (fchorney) [inline .github/workflows/main.yml:53]: I think I had this in the example I showed you. Possibly this was just for debugging purposes, and probably isn't needed here.

```suggestion
```

File: pyproject.toml [hunk 3]
```diff
@@ -1,2 +1,2 @@
 [tool.black]
-target-version = ["py37", "py38"]
+target-version = ["py38"]
```
Comment (fchorney) [inline pyproject.toml:2]:
```suggestion
target-version = ["py38", "py39"]
```

File: tox.ini [hunk 4]
```diff
@@ -1,22 +1,22 @@
...
-    py37,
     py38,
+    py39,
     coverage
     docs
 
 [testenv]
 usedevelop = True
 extras = test
+deps = Cython
```
Comment (fchorney) [inline tox.ini:12]: Why do we need `Cython` here? (Oh maybe this was from the example I showed you? In that case it was necessary for that specific thing)

**Comment by fchorney** [comment]
We might want to add the Slack notification stuff for this as well.

**Comment by nicoleepp** [comment]
> We might want to add the Slack notification stuff for this as well.

Nightly runs? My plan is to do that in a separate MR as a follow up unless there is a reason to do it together?

File: pyproject.toml (hunk 3 above)
Comment (nicoleepp) [reply pyproject.toml:2]: I tried this but py39 didn't work: https://github.com/invenia/TagBotGitLab/runs/1704491625?check_suite_focus=true

File: tox.ini (hunk 4 above)
Comment (nicoleepp) [reply tox.ini:12]: Should I add back the `basepython = python3` statements I removed?

**Comment by fchorney** [comment]
yeah I guess we can do it in a separate PR, I just figured since it was CI related we could have it here as well. Either works

File: pyproject.toml (hunk 3 above)
Comment (fchorney) [reply pyproject.toml:2]: ohh yeah sorry. I tried doing this in a personal project once and it didn't work also. I think black hasn't added that yet. Just keep it at `py38` only then.

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.x
+
+      # Install Deps
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install tox
+      # Run Check
```
Comment (fchorney) [inline .github/workflows/main.yml:23]: Aesthetics :P
```suggestion

      # Run Check
```

File: tox.ini (hunk 4 above)
Comment (fchorney) [reply tox.ini]: hmm possibly, or else when running on a local machine it would run `python` by default, which might still point to python 2

**Review by fchorney — APPROVED** [review]
LGTM

**Pull request merged**